### ISSUE TITLE
Feature: optional calibration_batch for INT8 export (decouple from export batch)

### ISSUE THREAD

**Issue by AffanBinFaisal** [ultralytics/ultralytics#24479]
## Summary

Decouple INT8 **calibration dataloader** batch size from export **`batch`** (dummy input / ONNX batch / TRT profile).

## Motivation / context

Question and maintainer reply in #24470: today `batch` is shared; workaround is `dynamic=True` with larger `batch`, which does not cover "static deploy batch = 1, calibrate with larger batch".

## Proposed API

- Add optional export kwarg, e.g. `calibration_batch: int | None` (default `None` → current behavior: use `batch`).

## Expected behavior

- **Default:** unchanged.
- **When set:** use only for `get_int8_calibration_dataloader` (and any backend that must match that batch, e.g. TensorRT calibrator / profile `opt` shape).

## Constraints (for implementation)

- TensorRT **static** batch: calibration batch must match network input batch → error or require `calibration_batch == batch`.
- TensorRT **`dynamic=True`:** allow `calibration_batch` consistent with optimization profile / `onnx2engine` shape so calibrator and bindings agree.

## Backends to consider

OpenVINO, TensorRT, TF SavedModel/TFLite calibration path, Axelera, IMX (any path using `get_int8_calibration_dataloader`).

Refs: #24470

**Comment by UltralyticsAssistant** [comment]
👋 Hello @AffanBinFaisal, thank you for your interest in Ultralytics 🚀! This is an automated response to help get your issue triaged, and an Ultralytics engineer will also assist soon.

Thanks for the clear feature proposal around decoupling INT8 calibration batch size from export `batch` 👍. We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python) and [CLI](https://docs.ultralytics.com/usage/cli) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips).

For this feature request, it would also help to share any relevant export command examples, target backend details, and deployment constraints so the team can evaluate the proposal across affected INT8 paths 🧩

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes) and [Tasks](https://docs.ultralytics.com/tasks) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by AffanBinFaisal** [comment]
I will open a PR as soon if you approve this task
@Y-T-G @glenn-jocher

**Comment by Y-T-G** [comment]
I don't there's a reason or benefit in decoupling them because I don't think you can calibrate with a batch size different from the model's batch size. And if you're using `dynamic=True`, then `batch` would just be the calibration batch size anyway, so there's no need for a separate argument 

https://github.com/ultralytics/ultralytics/issues/24470#issuecomment-4433895298

**Comment by AffanBinFaisal** [comment]
So there's no possible case where the user wants the calibration batch to be different than the model's batch size?

**Comment by Y-T-G** [comment]
I don't think so.

The recommendation about batch size is not applicable to Ultralytics because Ultralytics uses MINMAX_CALIBRATOR for TensorRT
